structured_prediction_baselines/models/multilabel_classification.py

```diff
@@ -83,3 +84,99 @@ def get_true_metrics(self, reset: bool = False) -> Dict[str, float]:
         }
 
         return metrics
+
+
+@Model.register(
+    "multi-label-classification-with-scorenn-evaluation",
+    constructor="from_partial_objects",
+)
+@Model.register(
+    "multi-label-classification-with-infnet-and-scorenn-evaluation",
+    constructor="from_partial_objects_with_shared_tasknn",
+)
+class MultilabelClassificationWithScoreNNEvaluation(MultilabelClassification):
+    def __init__(
+        self,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(**kwargs)
+        if self.evaluation_module is None:
+            raise ValueError("Evaluation Module can not be none for this sampler")
+
+        self.micro_map = MultilabelClassificationMicroAvgPrecision()
+        self.average_rank = MultilabelClassificationAvgRank()
+        self.mrr = MultilabelClassificationMeanReciprocalRank()
+        self.ndcg = MultilabelClassificationNormalizedDiscountedCumulativeGain()
+        self.rbo = MultilabelClassificationRankBiasedOverlap()
+        self.distribution_samples_f1 = MultilabelClassificationF1()
+        self.random_samples_f1 = MultilabelClassificationF1()
+
+    @torch.no_grad()
+    def calculate_metrics(  # type: ignore
+        self,
+        x: Any,
+        labels: torch.Tensor,
+        y_hat: torch.Tensor,
+        buffer: Dict,
+    ) -> None:
+
+        self.map(y_hat, labels)
```

Comment:
Instead of repeating code, we should call the super's calculate_metrics() in the beginning. This has two reasons behind it,
1. Metrics are volatile, meaning we might comment out or add metrics as need on the parent class. The child class' reliance on the attributes on self which are created in the parent's __init__(), like `self.map` is not safe. We might get errors if we switch off metrics.

2. If we add another metric for MLC, we don't have to add it to two files.